griptape/tools/google_drive/tool.py

```diff
@@ -0,0 +1,333 @@
+                        else:
+                            for a in artifacts:
+                                self._save_to_drive(f"{a.name}-{file_name}", a.value, folder_id)
+
+                        return InfoArtifact(f"saved successfully")
+
+                    except Exception as e:
+                        return ErrorArtifact(f"error saving file to Google Drive: {e}")
+                else:
+                    return ErrorArtifact(f"Could not find folder: {values['folder_path']}")
+            else:
+                return ErrorArtifact("no artifacts found")
+        else:
+            return ErrorArtifact("memory not found")
+
+    @activity(config={
+        "description": "Can be used to save content to a file on Google Drive",
+        "schema": Schema(
+            {
+                Literal(
+                    "path",
+                    description="Destination file path on Google Drive in the POSIX format. "
+                                "For example, 'foo/bar/baz.txt'"
+                ): str,
+                "content": str
+            }
+        )
+    })
+    def save_content_to_drive(self, params: dict) -> ErrorArtifact | InfoArtifact:
+        content = params["values"]["content"]
+        filename = params["values"]["path"]
+
+        try:
+            self._save_to_drive(filename, content)
+
+            return InfoArtifact(f"saved successfully")
+        except Exception as e:
+            return ErrorArtifact(f"error saving file to Google Drive: {e}")
+
+    @activity(config={
+        "description": "Can be used to download multiple files from Google Drive based on a provided list of paths",
+        "schema": Schema({
+            Literal(
+                "paths",
+                description="List of paths to files to be loaded in the POSIX format. "
+                            "For example, ['foo/bar/file1.txt', 'foo/bar/file2.txt']"
+            ): [str]
+        })
+    })
+    def download_files(self, params: dict) -> ListArtifact | ErrorArtifact:
+        values = params["values"]
+        downloaded_files = []
+    
+        service = self._build_client(self.LIST_FILES_SCOPES)
+    
+        for path in values["paths"]:
+            try:
+                file_id = self._path_to_file_id(service, path)
+                if file_id:
+                    file_info = service.files().get(fileId=file_id).execute()
+                    mime_type = file_info["mimeType"]
+    
+                    if mime_type in self.GOOGLE_EXPORT_MIME_MAPPING:
+                        export_mime = self.GOOGLE_EXPORT_MIME_MAPPING[mime_type]
+                        request = service.files().export_media(fileId=file_id, mimeType=export_mime)
+                    else:
+                        request = service.files().get_media(fileId=file_id)
+    
+                    downloaded_file = request.execute()
+                    downloaded_files.append(BlobArtifact(downloaded_file))
```

Comment:
If the variable is not re-used and if the statement on the right side of the assignment is short, we should aim to combine it in the same statement. In this case, it should look like this:

```python
downloaded_files.append(
    BlobArtifact(request.execute())
)
```

It improves readability.

Reply:
Also, just realized that it looks a bit sus. Wanted to make sure `execute()` returns a byte string instead of a JSON object.